### PR TITLE
lxd-metadata: Annotate codebase for `physical` network config keys

### PR DESCRIPTION
**Pull request by gabrielmougard**
See the root PR https://github.com/canonical/lxd/pull/13027 for a summary

### CODE REVIEW

**Comment by ru-fu** [comment]
I added my commits.
This will now fail the doc build, since I added references to device options - so we'll need a rebase.

**Comment by gabrielmougard** [comment]
@ru-fu updated

**Comment by tomponline** [comment]
please rebase

File: lxd/network/driver_physical.go [hunk 1]
```diff
@@ -29,26 +29,158 @@ func (n *physical) DBType() db.NetworkType {
...
+		//  shortdesc: IPv4 ranges to use for child OVN network routers
+		"ipv4.ovn.ranges": validate.Optional(validate.IsListOf(validate.IsNetworkRangeV4)),
+		// lxdmeta:generate(entities=network-physical; group=network-conf; key=ipv6.ovn.ranges)
+		// Specify a comma-separated list of IPv6 ranges in FIRST-LAST format.
+		// ---
+		//  type: string
+		//  shortdesc: IPv6 ranges to use for child OVN network routers
+		"ipv6.ovn.ranges": validate.Optional(validate.IsListOf(validate.IsNetworkRangeV6)),
+		// lxdmeta:generate(entities=network-physical; group=network-conf; key=ipv4.routes)
+		// Specify a comma-separated list of IPv4 CIDR subnets that can be used with the child OVN network's {config:option}`network-bridge-network-conf:ipv4.routes.external` setting.
```
Comment (ru-fu) [inline lxd/network/driver_physical.go:98]:
```suggestion
		// Specify a comma-separated list of IPv4 CIDR subnets that can be used with the child OVN network's {config:option}`device-nic-ovn-device-conf:ipv4.routes.external` setting.
```

No idea how I managed to copy&paste this link here ... I actually wrote that I'm linking to the device option. :smile: 
Let's hope this works now.

File: lxd/network/driver_physical.go [hunk 2]
```diff
@@ -29,26 +29,158 @@ func (n *physical) DBType() db.NetworkType {
...
+		// lxdmeta:generate(entities=network-physical; group=network-conf; key=ipv4.routes.anycast)
+		// If set to `true`, this option allows the overlapping routes to be used on multiple networks/NICs at the same time.
+		// ---
+		//  type: bool
+		//  condition: IPv4 address
+		//  defaultdesc: `false`
+		//  shortdesc: Whether to allow IPv4 routes on multiple networks/NICs
+		"ipv4.routes.anycast": validate.Optional(validate.IsBool),
+		// lxdmeta:generate(entities=network-physical; group=network-conf; key=ipv6.routes)
+		// Specify a comma-separated list of IPv6 CIDR subnets that can be used with the child OVN network's {config:option}`network-bridge-network-conf:ipv6.routes.external` setting.
```
Comment (ru-fu) [inline lxd/network/driver_physical.go:113]:
```suggestion
		// Specify a comma-separated list of IPv6 CIDR subnets that can be used with the child OVN network's {config:option}`device-nic-ovn-device-conf:ipv6.routes.external` setting.
```

File: lxd/network/driver_physical.go (hunk 1 above)
Comment (gabrielmougard) [reply lxd/network/driver_physical.go]: fixed

File: lxd/network/driver_physical.go (hunk 2 above)
Comment (gabrielmougard) [reply lxd/network/driver_physical.go]: fixed

**Review by ru-fu — APPROVED** [review]

**Review by tomponline — APPROVED** [review]

**Pull request merged**